nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/kinesis/firehose/ITPutKinesisFirehose.java

```diff
@@ -0,0 +1,149 @@
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * This test contains both unit and integration test (integration tests are ignored by default)
+ */
+public class ITPutKinesisFirehose {
+
+    private TestRunner runner;
+    protected final static String CREDENTIALS_FILE = System.getProperty("user.home") + "/aws-credentials.properties";
+
+    @Before
+    public void setUp() throws Exception {
+        runner = TestRunners.newTestRunner(PutKinesisFirehose.class);
+        runner.setProperty(PutKinesisFirehose.ACCESS_KEY, "abcd");
+        runner.setProperty(PutKinesisFirehose.SECRET_KEY, "secret key");
+        runner.setProperty(PutKinesisFirehose.KINESIS_FIREHOSE_DELIVERY_STREAM_NAME, "deliveryName");
+        runner.assertValid();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        runner = null;
+    }
+
+    @Test
+    public void testCustomValidateBatchSize1Valid() {
+        runner.setProperty(PutKinesisFirehose.BATCH_SIZE, "1");
+        runner.assertValid();
+    }
+
+    @Test
+    public void testCustomValidateBatchSize500Valid() {
+        runner.setProperty(PutKinesisFirehose.BATCH_SIZE, "500");
+        runner.assertValid();
+    }
+    @Test
+    public void testCustomValidateBatchSize501InValid() {
+        runner.setProperty(PutKinesisFirehose.BATCH_SIZE, "501");
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testCustomValidateBufferSize1Valid() {
+        runner.setProperty(PutKinesisFirehose.MAX_BUFFER_SIZE, "1");
+        runner.assertValid();
+    }
+
+    @Test
+    public void testCustomValidateBufferSize128Valid() {
+        runner.setProperty(PutKinesisFirehose.MAX_BUFFER_SIZE, "128");
+        runner.assertValid();
+    }
+    @Test
+    public void testCustomValidateBufferSize129InValid() {
+        runner.setProperty(PutKinesisFirehose.MAX_BUFFER_SIZE, "129");
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testCustomValidateBufferInterval900Valid() {
+        runner.setProperty(PutKinesisFirehose.MAX_BUFFER_INTERVAL, "900");
+        runner.assertValid();
+    }
+
+    @Test
+    public void testCustomValidateBufferInterval60Valid() {
+        runner.setProperty(PutKinesisFirehose.MAX_BUFFER_INTERVAL, "60");
+        runner.assertValid();
+    }
+
+    @Test
+    public void testCustomValidateBufferInterval901InValid() {
+        runner.setProperty(PutKinesisFirehose.MAX_BUFFER_INTERVAL, "901");
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testCustomValidateBufferInterval59InValid() {
+        runner.setProperty(PutKinesisFirehose.MAX_BUFFER_INTERVAL, "59");
+        runner.assertNotValid();
+    }
+
+    /**
+     * Comment out ignore for integration tests (requires creds files)
+     */
+    @Test
+    @Ignore
```

Comment:
The profile to run ITs is not enabled by default and this Ignore is unneeded.  Would prefer to have removed so that no code changes are needed to run them.